tests/_support/_bootstrap.php

```diff
@@ -24,5 +24,5 @@
 //--------------------------------------------------------------------
 // Load our TestCase
 //--------------------------------------------------------------------
-
+require_once (TESTPATH . '/../vendor/autoload.php');
```

Comment:
Were you running into a case where autoload wasn't working here? Should already be included, I believe.

Reply:
Hello Lonnie,

I have been a long time user of CodeIgniter from the 1.x and 2.x days and I'm glad to see version 4 being put together. I would like to give back and volunteer: since the tests are a "help wanted" item I thought that I could start here.

TL:DR:

After further testing I realized that this is not necessary if you run from the composer installed version of phpUnit.  The line is necessary if your are running from a `phpunit.phar`.

If you would rather not change the code here I could add a section to the readme about how to get things to work with `phpunit.phar`.

Details:

This change allows you to run your code with minimal bootstrapping of the system: you aren't running the entire system unless you choose to but you still can extend `CIUnitTestCase`.  If you comment that out, the code below will fail.  This is running from a phpunit.phar file. 

When the tests run from the code igniter root directory as described in `tests/README.md` it picks up the autoloader file by searching in the vendor directory automagically.

PHPStorm has a helper for this and you can also run the code generated from it from the linux console this way (alter for your own path, this is running from phpunit.phar at /home/vagrant/Code/):

The test from a phpunit.phar file (needs the autoloader line):

`/home/vagrant/Code/phpunit --bootstrap /home/vagrant/Code/codeigniter4/tests/_support/_bootstrap.php --no-configuration BootstrapFCPATHTest /home/vagrant/Code/codeigniter4/tests/system/Test/BootstrapFCPATHTest.php --teamcity`

The test uses the composer installed version (does not need the autoloader line):

`/home/vagrant/Code/codeigniter4/vendor/phpunit/phpunit/phpunit --bootstrap /home/vagrant/Code/codeigniter4/tests/_support/_bootstrap.php --no-configuration CodeIgniter\\Cache\\CacheFactoryTest /home/vagrant/Code/codeigniter4/tests/system/Cache/CacheFactoryTest.php --teamcity`

The output looks different but the tests pass.


Cheers,
Chris

Reply:
Hey Chris - definitely appreciate you wanting to jump in and help out! Definitely don't want to discourage that. 

In this case, I think the doc change is best, and leaving that line out. I've found it's generally better to not use a global version and to use a composer-installed version since different versions of frameworks can demand different versions PHPunit to work properly.

Reply:
Ok I'll work on that instead and possibly make a blog post or some other documentation for alternatives/setting up PHPStorm for testing.